paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -76,7 +79,11 @@ class PaparazziPlugin : Plugin<Project> {
         project.tasks.named("merge${variantSlug}Assets") as TaskProvider<MergeSourceSetFolders>
       val mergeAssetsOutputDir = mergeAssetsProvider.flatMap { it.outputDir }
       val reportOutputDir = project.layout.buildDirectory.dir("reports/paparazzi")
-      val snapshotOutputDir = project.layout.projectDirectory.dir("src/test/snapshots")
+
+      val snapshotOutputDir =
+          extension.snapshotRootDirectory.orNull
+          ?: project.layout.projectDirectory.dir("src/test/snapshots").asFile
+      snapshotOutputDir.mkdirs()
```

Comment:
Seems like in PrepareResourcesTask to add this as an `InputDirectory` the directory has to exist, so in that case I have to add this to create the snapshot directory here instead of waiting until the task to take/verify the snapshots. Thoughts on if this is fine?

Reply:
you went from a Provider to a Provider that's resolved in configuration phase. This is happening too early here I think. I think this would be something like:
```
extension.snapshotRootDirectory.orElse(project.layout.projectDirectory.dir("src/test/snapshots"))
```
and maybe that solves your need to create the dir too.

Reply:
but it might be even better if the default was set up with `snapshotRootDirectory.convention` when the extension is registered:
```
val extension = project.extensions.create("paparazzi", PaparazziPluginExtension::class.java)
extension.snapshotRootDirectory.convention(project.layout.projectDirectory.dir("src/test/snapshots"))
```